src/redisearch_rs/build_utils/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! build.rs utilities.
+
+use std::{fs::read_dir, path::Path};
+
+/// Return the root folder of the project containing the `.git` directory.
+pub fn git_root() -> std::path::PathBuf {
+    let mut path = std::env::current_dir().unwrap();
+    while !path.join(".git").exists() {
+        path = path.parent().unwrap().to_path_buf();
+    }
+    path
+}
```

Comment:
We have different variants of these floating around. I like the ones which don't `unwrap()` as we can use it to make the error clearer

https://github.com/RediSearch/RediSearch/blob/45c02e43d26c24d96c9fbd8959db4fb7cc7f483d/src/redisearch_rs/inverted_index_bencher/build.rs#L108-L117